components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java

```diff
@@ -58,12 +243,153 @@ public void createUserAccountAssociation(String userName, char [] password) thro
      * @throws org.wso2.carbon.identity.user.account.association.exception.UserAccountAssociationClientException
      */
     public void deleteUserAccountAssociation(String userName) throws UserAccountAssociationClientException {
+
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(userName);
+        int tenantId = MultitenantConstants.INVALID_TENANT_ID;
+        RealmService realmService;
+
+        try {
+            realmService = IdentityAccountAssociationServiceComponent.getRealmService();
+            tenantId = realmService.getTenantManager().getTenantId(MultitenantUtils.getTenantDomain(userName));
+        } catch (UserStoreException e) {
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .ERROR_WHILE_GETTING_TENANT_ID
+                    .getDescription(), e);
+        } catch (Exception e) {
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .ERROR_WHILE_LOADING_REALM_SERVICE
+                    .getDescription(), e);
+        }
+        String domainName = UserAccountAssociationUtil.getDomainName(tenantAwareUsername);
+        tenantAwareUsername = UserAccountAssociationUtil.getUsernameWithoutDomain(tenantAwareUsername);
+        try {
+            if (!UserAccountAssociationDAO.getInstance().isValidUserAssociation(domainName, tenantId,
+                    tenantAwareUsername)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(UserAccountAssociationConstants.ErrorMessages.INVALID_ASSOCIATION.getDescription());
+
+                }
+                throw new UserAccountAssociationClientException(UserAccountAssociationConstants
+                        .ErrorMessages.INVALID_ASSOCIATION.toString());
+            }
+            UserAccountConnectorImpl.getInstance().deleteUserAccountAssociation(userName);
+        } catch (UserAccountAssociationServerException e) {
+            log.error(UserAccountAssociationConstants.ErrorMessages.CONN_DELETE_ERROR.getDescription(), e);
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .CONN_DELETE_ERROR.toString());
```

Comment:
This is a admin service, here I have logged server exception without sending it to the client side and throw a client exception with user friendly error code and message.